### PR TITLE
[FLINK-5692] [config] Add an Option to Deactivate Kryo Fallback for Serializers

### PR DESCRIPTION
**Pull request by jinmingjian**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/test/java/org/apache/flink/api/common/ExecutionConfigTest.java [hunk 1]
```diff
@@ -64,4 +68,15 @@ public void testConfigurationOfParallelism() {
 		assertEquals(parallelism, config.getParallelism());
 	}
 
+	@Test(expected = UnsupportedOperationException.class)
```
Comment (StephanEwen) [inline flink-core/src/test/java/org/apache/flink/api/common/ExecutionConfigTest.java]: I think this test would also pass of line 75 already throws the exception.

I personally discourage the `@Test(expected = )` pattern completely, as you have no check which statement actually throws the exception.

**Comment by jinmingjian** [comment]
revision done. _"disableGenericTypes"_ makes good sense from the view of docs (although "generic types" tends to be overused).

File: flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java [hunk 2]
```diff
@@ -109,6 +109,8 @@
 
 	private boolean forceKryo = false;
 
+	private boolean disableGenericTypes = false;//ISSUE FLINK-5692
```
Comment (StephanEwen) [inline flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java]: We usually do not add the issues as comments to the code. The git history stores that (commits have the issue number).

**Comment by StephanEwen** [comment]
If you can remove the comment, that would be great. Otherwise we can merge this...

**Comment by jinmingjian** [comment]
@StephanEwen Just my coding habit. Correction done. 

And very appreciated for your review. I am open for more contribution! :tada:

**Comment by StephanEwen** [comment]
The change looks good, merging this...

**Pull request closed without merging**